node/channel.js

```diff
@@ -453,20 +454,37 @@ TChannel.prototype.makeSubChannel = function makeSubChannel(options) {
     return chan;
 };
 
-TChannel.prototype.listen = function listen(port, host, callback) {
+TChannel.prototype.listen = function listen(port, host, options, callback) {
     // Note:
     // - 0 is a valid port number, indicating that the system must assign an
     //   available ephemeral port
     // - 127.0.0.1 is a valid host, primarily for testing
+    // - options supports:
+    //   {fd: X} - listen will attach to the existing socket, and host/port
+    //   remain required, but become only descriptive. See also:
+    //   https://nodejs.org/api/net.html#net_server_listen_handle_callback
     var self = this;
+    if (options && typeof options === 'function') {
+        callback = options;
+        options = {};
+    }
+    options = options || {};
     assert(!self.listened, 'TChannel can only listen once');
     assert(typeof host === 'string', 'TChannel requires host argument');
     assert(typeof port === 'number', 'TChannel must listen with numeric port');
+    assert(options.fd === undefined || typeof options.fd === 'number', 'TChannel listen options.fd must be numeric');
```

Comment:
`typeof X === 'number' && X >= 0` should suffice here... and if someone passes a non-integer, well they'll get an exception anyhow when we try to pass it to listen:
- typeof undefined !== 'number'
- ditto for typeof null
- NaN annihilates all positive binary relations on numbers (i.e. X R NaN === false for R in <=, <, >, >=, ...)

In fact, simply the typeof should be fine, since NaN should blow up the underlying api anyhow, which is as good an assertion imo.